prow/git/git.go

```diff
@@ -138,7 +142,11 @@ func (c *Client) Clone(repo string) (*Repo, error) {
 	base := c.base
 	user, pass := c.getCredentials()
 	if user != "" && pass != "" {
```

Comment:
Yuck. This is extant code that _only_ handles testing cases and totally makes this PR not tested by any of the normal means

Reply:
We should not allow bad values like that used for testing in production clients

Reply:
if only we had some way of knowing which lines of code were covered by our tests